uctMaths/competition/compadmin.py

```diff
@@ -1288,7 +1296,6 @@ def get_student_answer_sheet(request, student):
         template = get_template('pair_as_template.html')
     else:
         template = get_template('individual_as_template.html')
-    c.update(csrf(request))
```

Comment:
Why did you remove this line?

Reply:
When done from the server side, the request is null, so it breaks completely when csrf is run with None

Reply:
I think it would be better to write

``` python
if request is not None:
    c.update(csrf(request))
```

I'm suprised we haven't. We haven't caught this ealier. This would cause the function  `bg_generate_as_grade_distinction` to fail.

Reply:
I did `if request:`, which also seems to work fine

Reply:
Actually this is fine, we don't need CSRF.